src/Microsoft.Framework.FileSystemGlobbing/Internal/PathSegments/LiteralPathSegment.cs

```diff
@@ -7,7 +7,9 @@ namespace Microsoft.Framework.FileSystemGlobbing.Internal.PathSegments
 {
     public class LiteralPathSegment : IPathSegment
     {
-        public LiteralPathSegment(string value)
+        private readonly StringComparison _comparisonType;
+
+        public LiteralPathSegment(string value, StringComparison comparisonType)
```

Comment:
It makes all the methods signature very verbose, if you track the flow of the StringComparison.